apps/api/src/app/shared/services/in-memory-provider/in-memory-provider.service.ts

```diff
@@ -1,41 +1,154 @@
+  getRedisProviderConfig,
+  IRedisProviderConfig,
+  Redis,
+  RedisOptions,
+} from './redis-provider';
+import {
+  Cluster,
+  ClusterOptions,
+  getRedisCluster,
+  getRedisClusterProviderConfig,
+  IRedisClusterProviderConfig,
+} from './redis-cluster-provider';
 
 import { ApiException } from '../../exceptions/api.exception';
 
 const LOG_CONTEXT = 'InMemoryCluster';
 
-export type InMemoryProviderClient = Redis | undefined;
-type InMemoryProviderConfig = IRedisProviderConfig;
+export type InMemoryProviderClient = Redis | Cluster | undefined;
+type InMemoryProviderConfig = IElasticacheClusterProviderConfig | IRedisProviderConfig | IRedisClusterProviderConfig;
 
+@Injectable()
 export class InMemoryProviderService {
   public inMemoryProviderClient: InMemoryProviderClient;
   public inMemoryProviderConfig: InMemoryProviderConfig;
 
   constructor() {
-    if (!this.inMemoryProviderClient) {
-      this.inMemoryProviderClient = process.env.IN_MEMORY_CLUSTER_MODE_ENABLED
-        ? this.inMemoryClusterProviderSetup()
-        : this.inMemoryProviderSetup();
-    }
+    this.inMemoryProviderClient = this.buildClient();
+  }
+
+  private buildClient(): Redis | Cluster | undefined {
+    return this.isClusterMode() ? this.inMemoryClusterProviderSetup() : this.inMemoryProviderSetup();
   }
 
   public isClientReady(): boolean {
     if (!this.inMemoryProviderClient) {
       return false;
     }
 
+    // TODO: Check this works for Cluster
     return this.inMemoryProviderClient.status === CLIENT_READY;
   }
 
-  private inMemoryClusterProviderSetup(): InMemoryProviderClient {
+  public isClusterMode(): boolean {
+    return process.env.IN_MEMORY_CLUSTER_MODE_ENABLED === 'true';
+  }
+
+  public getClusterOptions(): ClusterOptions | undefined {
+    if (this.inMemoryProviderClient && this.isClusterMode()) {
+      return this.inMemoryProviderClient.options;
+    }
+  }
+
+  public getOptions(): RedisOptions | undefined {
+    if (this.inMemoryProviderClient) {
+      if (this.isClusterMode()) {
+        const clusterOptions: ClusterOptions = this.inMemoryProviderClient.options;
+
+        return clusterOptions.redisOptions;
+      } else {
+        const options: RedisOptions = this.inMemoryProviderClient.options;
+
+        return options;
+      }
+    }
+  }
+
+  private isElasticacheEnabled(): boolean {
+    return !!process.env.ELASTICACHE_CLUSTER_SERVICE_HOST && !!process.env.ELASTICACHE_CLUSTER_SERVICE_PORT;
+  }
+
+  private getClientAndConfigForCluster(): {
+    getClient: () => Cluster | undefined;
+    getConfig: () => InMemoryProviderConfig;
+  } {
+    const clusterProviders = {
+      elasticache: {
+        getClient: getElasticacheCluster,
+        getConfig: getElasticacheClusterProviderConfig,
+      },
+      redis: {
+        getClient: getRedisCluster,
+        getConfig: getRedisClusterProviderConfig,
+      },
+    };
+
+    return this.isElasticacheEnabled() ? clusterProviders.elasticache : clusterProviders.redis;
```

Comment:
If Elasticache configuration is set in the environment we use that provider and if not we fall back to the Redis Cluster provider that needs automated setting in the CI/CD and in your local environment when running the tests. If the Cluster mode is enabled in the configuration through `IN_MEMORY_CLUSTER_MODE_ENABLED`.